files/acl.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Automattic\VIP\Files\Acl;
+
+const FILE_IS_PUBLIC = 'FILE_IS_PUBLIC';
+const FILE_IS_PRIVATE_AND_ALLOWED = 'FILE_IS_PRIVATE_AND_ALLOWED';
+const FILE_IS_PRIVATE_AND_DENIED = 'FILE_IS_PRIVATE_AND_DENIED';
+const FILE_NOT_FOUND = 'FILE_NOT_FOUND';
+
+function send_visibility_headers( $file_visibility, $file_path ) {
+	// Default to throwing an error so we can catch unexpected problems more easily.
+	$status_code = 500;
+	$header = false;
+
+	switch ( $file_visibility ) {
+		case FILE_IS_PUBLIC:
+			$status_code = 202;
+			break;
+
+		case FILE_IS_PRIVATE_AND_ALLOWED:
+			$status_code = 202;
+			$header = 'X-Private: true';
+			break;
+
+		case FILE_IS_PRIVATE_AND_DENIED:
+			$status_code = 403;
+			$header = 'X-Private: true';
+			break;
+
+		case FILE_NOT_FOUND:
+			// This is a signal for the server to pass the request through to the Files Service.
+			// Even though the attachment was not found within WordPress, it may still be a physical file in the Files Service.
+			$status_code = 202;
+			break;
+
+		default:
+			// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+			trigger_error( sprintf( 'Invalid file visibility (%s) ACL set for %s', $file_visibility, $file_path ), E_USER_WARNING );
+			break;
+	}
+
+	http_response_code( $status_code );
+
+	if ( $header ) {
+		header( $header );
+	}
+}
+
+/**
+ * Variant of core's attachment_url_to_postid() function
+ *
+ * @param $path (string) The path to resolve.
+ *
+ * @return (int) The found attachment ID, or 0 on failure.
+ */
+function get_file_path_from_attachment_id( $path ) {
```

Comment:
It might make sense to move this out and include it with the "restrict media on unpublished content" PR since it's only going to be used there.

Reply:
Moved to https://gist.github.com/mjangda/8657cfde2b2a4cafd95f509dd1d3dde9#file-vip-private-files-unpublished-media-php-L42